Local development environment not producable for contribution

### Describe the bug

Hi there,

I'm want to contribute to Houdini however the development environment does not function with the instructions provided on the `[website](https://www.houdinigraphql.com/guides/contributing)`.

If this is of any help: I am on Arch Linux (EndeavorOS).

### Severity

blocking all usage of Houdini

### Steps to Reproduce the Bug

1. Clone the houdini repo
2. Follow instructions on [guides/contributing](https://www.houdinigraphql.com/guides/contributing)
3. `pnpm i && pnpm run build`
4. This gets peer dependency errors (can be ignored with `strict-peer-dependencies=false`). Build also fails.
6. `cd example`
7. `pnpm run generate && pnpm run dev`
8. generate fails and dev fails because of this.



### Reproduction

_No response_

The example fails to build for me also.

Build error:
```
houdini/example/src/routes/[filter]/+page.svelte:4:23
 2 |  import GQL_AddItem from "$houdini/stores/AddItem";
 3 |  import { page } from "$app/stores";
 4 |  import { graphql, type AddItemStore, type AllItemsStore, type NewItemStore } from "$houdini";
                             ^
 5 |  import ItemEntry from "$lib/ItemEntry.svelte";
 6 |  import { derived } from "svelte/store";
error during build:
ParseError: Unexpected token
    at error (file:///houdini/node_modules/.pnpm/svelte@3.49.0/node_modules/svelte/compiler.mjs:17799:19)
```

Dev error:
```
TypeError: __vite_ssr_import_1__.default.init is not a function
    at eval (/src/routes/+layout.js:18:31)
```

I can run the build successfully, but I noticed this in the output:
```
[cmd] [exe] rpt2: options error TS6053: File 'houdini/example/$houdini/graphql/enums.d.ts' not found.
[cmd] [exe]   The file is in the program because:
[cmd] [exe]     Root file specified for compilation
```

> @sjcobb2022 I had an issue with the integration tests. But using `nvm` to switch to `node v16.17.0` fixed it.
> 
> Just curious, what version of `pnpm` and `node` are you using. Also, can you post a gist with your error ?

Hi @DanielHritcu. 

Sorry for the late response I'm prepping for uni. I am on node v18.9.0 and pnpm v7.12.1.

The errors appear as follows

## Using node v18.9.0

pnpm i:
```
ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies

example
└─┬ graphql-relay 0.8.0
  └── ✕ unmet peer graphql@15.5.1: found 15.5.0

integration
├─┬ @replayio/playwright 0.2.23
│ └── ✕ unmet peer @playwright/test@1.19.x: found 1.25.0
└─┬ graphql-relay 0.10.0
  └── ✕ unmet peer graphql@^16.2.0: found 15.5.0

site
└─┬ vite-plugin-replace 0.1.1
  └── ✕ unmet peer vite@^2: found 3.0.9
```
pnpm generate:
```
> example-kit@0.13.0 generate /home/sjcobb/dev/houdini/example
> houdini generate

sh: line 1: houdini: command not found
 ELIFECYCLE  Command failed.
```
then for pnpm run dev

when running it is clear that sveltekit is clearly not initialized. (there is not .svelte-kit when I list the directory)

```
[web] ▲ [WARNING] Cannot find base config file "./.svelte-kit/tsconfig.json" [tsconfig.json]
``` 

The graphQL server with yoga then also proceeds to fail it's init.

```
[api] 💡 🧘 Yoga - Running GraphQL Server at http://0.0.0.0:4000/graphql
[web] ⚠️  Couldn't pull your latest schema: request to http://localhost:4000/graphql failed, reason: connect ECONNREFUSED ::1:4000
```

And then when trying to access localhost

```
[web] __vite_ssr_import_1__.default.init is not a function
[web] TypeError: __vite_ssr_import_1__.default.init is not a function
[web]     at /src/routes/+layout.js:13:15
```


## Using Node 

pnpm i
```
 ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies

example
└─┬ graphql-relay 0.8.0
  └── ✕ unmet peer graphql@15.5.1: found 15.5.0

integration
├─┬ @replayio/playwright 0.2.23
│ └── ✕ unmet peer @playwright/test@1.19.x: found 1.25.0
└─┬ graphql-relay 0.10.0
  └── ✕ unmet peer graphql@^16.2.0: found 15.5.0

site
└─┬ vite-plugin-replace 0.1.1
  └── ✕ unmet peer vite@^2: found 3.0.9

```

pnpm run dev: 
```
[web] ▲ [WARNING] Cannot find base config file "./.svelte-kit/tsconfig.json" [tsconfig.json]

```

accessing localhost: 
```
[web] __vite_ssr_import_1__.default.init is not a function
[web] TypeError: __vite_ssr_import_1__.default.init is not a function
[web]     at /src/routes/+layout.js:13:15
```

So the only difference is that the yoga server is queried correctly.

--- a/.changeset/large-forks-kneel.md
+++ b/.changeset/large-forks-kneel.md
@@ -1,0 +1,5 @@
+---
+'houdini': patch
+---
+
+Mutation result is never null

--- a/.changeset/sour-students-juggle.md
+++ b/.changeset/sour-students-juggle.md
@@ -1,0 +1,5 @@
+---
+'houdini': patch
+---
+
+Subscription.listen arguments are optional

--- a/site/src/routes/api/mutation.svx
+++ b/site/src/routes/api/mutation.svx
@@ -122,7 +122,7 @@ and provide an "optimistic response" for the mutation with the second argument t
 
 	export let itemID
 
-	const store: ToggleItemStore = graphql`
+	const toggleItem: ToggleItemStore = graphql`
 		mutation ToggleItem($id: ID!) {
 			toggleItem {
 				item {

--- a/site/src/routes/guides/working-with-graphql.svx
+++ b/site/src/routes/guides/working-with-graphql.svx
@@ -35,8 +35,8 @@ const mutate = mutation(graphql\`
     const externalMutationGraphQL = `
 # src/lib/graphql/UpdateUser.gql
 
-mutation UpdateUser($input: UpdateUserInput!) {
-    updateUser(input: $input) {
+mutation UpdateUser($firstName: String!) {
+    updateUser(firstName: $firstName) {
         user {
             firstName
         }
@@ -49,7 +49,9 @@ mutation UpdateUser($input: UpdateUserInput!) {
 
 import { GQL_UpdateUser } from '$houdini'
 
-const mutate = GQL_UpdateUser.mutate
+async function update() {
+    await GQL_UpdateUser.mutate({ firstName: "..." } )
+}
 `
 
 </script>

--- a/src/runtime/stores/mutation.ts
+++ b/src/runtime/stores/mutation.ts
@@ -35,7 +35,7 @@ export class MutationStore<
 			metadata?: App.Metadata
 			fetch?: typeof globalThis.fetch
 		} & MutationConfig<_Data, _Input, _Optimistic> = {}
-	): Promise<_Data | null> {
+	): Promise<_Data> {
 		const config = await this.getConfig()
 
 		this.store.update((c) => {

--- a/src/runtime/stores/mutation.ts
+++ b/src/runtime/stores/mutation.ts
@@ -132,7 +132,7 @@ export class MutationStore<
 			this.store.set(storeData)
 
 			// return the value to the caller
-			return storeData.data
+			return storeData.data ?? ({} as _Data)
 		} catch (error) {
 			this.store.update((s) => ({
 				...s,

--- a/src/runtime/stores/subscription.ts
+++ b/src/runtime/stores/subscription.ts
@@ -28,10 +28,10 @@ export class SubscriptionStore<_Data, _Input extends {}> extends BaseStore {
 		return this.store?.subscribe(...args)
 	}
 
-	async listen(variables: _Input) {
+	async listen(variables?: _Input) {
 		// @ts-expect-error: typechecking cjs/esm interop is hard
 		// pull the query text out of the compiled artifact
-		const { raw: text, selection } = artifact.default || artifact
+		const { raw: text, selection } = this.artifact.default || this.artifact
 
 		// subscription.listen is a no-op on the server
 		if (!isBrowser) {

